homeassistant/components/google.py

```diff
@@ -0,0 +1,294 @@
+"""
+Support for Google - Calendar Event Devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/google/
+
+NOTE TO OTHER DEVELOPERS: IF YOU ADD MORE SCOPES TO THE OAUTH THAN JUST
+CALENDAR THEN USERS WILL NEED TO DELETE THEIR TOKEN_FILE. THEY WILL LOSE THEIR
+REFRESH_TOKEN PIECE WHEN RE-AUTHENTICATING TO ADD MORE API ACCESS
+IT'S BEST TO JUST HAVE SEPARATE OAUTH FOR DIFFERENT PIECES OF GOOGLE
+"""
+import logging
+import os
+import threading
+import yaml
+
+import voluptuous as vol
+from voluptuous.error import Error as VoluptuousError
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.loader as loader
+from homeassistant import bootstrap
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.event import track_time_change
+from homeassistant.util import convert, dt
+
+REQUIREMENTS = [
+    'google-api-python-client==1.5.5',
+    'oauth2client==3.0.0',
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'google'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONF_TRACK_NEW = 'track_new_calendar'
+
+CONF_CAL_ID = 'cal_id'
+CONF_DEVICE_ID = 'device_id'
+CONF_NAME = 'name'
+CONF_ENTITIES = 'entities'
+CONF_TRACK = 'track'
+CONF_SEARCH = 'search'
+CONF_OFFSET = 'offset'
+
+DEFAULT_CONF_TRACK_NEW = True
+DEFAULT_CONF_OFFSET = '!!'
+
+NOTIFICATION_ID = 'google_calendar_notification'
+NOTIFICATION_TITLE = 'Google Calendar Setup'
+GROUP_NAME_ALL_CALENDARS = "Google Calendar Sensors"
+
+SERVICE_SCAN_CALENDARS = 'scan_for_calendars'
+SERVICE_FOUND_CALENDARS = 'found_calendar'
+
+DATA_INDEX = 'google_calendars'
+
+YAML_DEVICES = '{}_calendars.yaml'.format(DOMAIN)
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+
+TOKEN_FILE = '.{}.token'.format(DOMAIN)
+
+PLATFORM_SCHEMA = vol.Schema({
```

Comment:
This is a component, so you should use `CONFIG_SCHEMA`. See [group example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/group.py#L58-L65)